src/code42cli/cmds/devices.py

```diff
@@ -0,0 +1,449 @@
+
+
+def _deactivate_device(sdk, device_guid, change_device_name, purge_date):
+    device = sdk.devices.get_by_guid(device_guid)
+    try:
+        sdk.devices.deactivate(device.data["computerId"])
+    except exceptions.Py42BadRequestError:
+        raise Code42CLIError("The device {} is in legal hold.".format(device_guid))
+    except exceptions.Py42NotFoundError:
+        raise Code42CLIError("The device {} was not found.".format(device_guid))
+    except exceptions.Py42ForbiddenError:
+        raise Code42CLIError("Unable to deactivate {}.".format(device_guid))
+    if purge_date:
+        _update_cold_storage_purge_date(sdk, device_guid, purge_date)
+    if change_device_name and not device.data["name"].startswith("deactivated_"):
+        _change_device_name(
+            sdk,
+            device_guid,
+            "deactivated_"
+            + date.today().strftime("%Y-%m-%d")
+            + "_"
+            + device.data["name"],
+        )
+
+
+def _update_cold_storage_purge_date(sdk, guid, purge_date):
+    archives_response = sdk.archive.get_all_by_device_guid(guid)
+    archive_guid_list = [
+        archive["archiveGuid"]
+        for page in archives_response
+        for archive in page["archives"]
+        if archive["format"] != "ARCHIVE_V2"
+    ]
+    for archive_guid in archive_guid_list:
+        sdk.archive.update_cold_storage_purge_date(
+            archive_guid, purge_date.strftime("%Y-%m-%d")
+        )
+
+
+def _change_device_name(sdk, guid, name):
+    device_settings = sdk.devices.get_settings(guid)
+    device_settings.name = name
+    sdk.devices.update_settings(device_settings)
+
+
+@devices.command()
+@device_guid_argument
+@format_option
+@sdk_options()
+def show(state, device_guid, format=None):
+    """Print device info. Requires device GUID."""
+
+    formatter = OutputFormatter(format, _device_info_keys_map())
+    backup_set_formatter = OutputFormatter(format, _backup_set_keys_map())
+    device_info = _get_device_info(state.sdk, device_guid)
+    formatter.echo_formatted_list([device_info])
+    click.echo()
+    backup_set_formatter.echo_formatted_list(device_info["backupUsage"])
+
+
+def _device_info_keys_map():
+    return {
+        "name": "Name",
+        "osHostname": "Hostname",
+        "guid": "GUID",
+        "status": "Status",
+        "lastConnected": "Last Connected Date",
+        "productVersion": "Code42 Version",
+        "osName": "Operating System",
+        "osVersion": "Operating System Version",
+    }
+
+
+def _backup_set_keys_map():
+    return {
+        "targetComputerName": "Destination",
+        "lastBackup": "Last Backup Activity",
+        "lastCompleted": "Last Completed Backup",
+        "archiveBytes": "Archive Size in Bytes",
+        "archiveGuid": "Archive GUID",
+    }
+
+
+def _get_device_info(sdk, device_guid):
+    return sdk.devices.get_by_guid(device_guid, include_backup_usage=True).data
+
+
+def _get_key_from_list_of_dicts(key, list_of_dicts):
```

Comment:
Looks like this is no longer used anywhere.

Reply:
Removed.